### PR TITLE
Card: Initial implementation

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR contains the first implementation of a `Card` component using our latest stack of technologies.

image

#### Notes

- It's using `office-ui-fabric-react` components for the examples.
- There are a couple of TODOs regarding some props that I still need to address.

### CODE REVIEW

**Comment by khmakoto** [comment]
There's an open question about what to do with the `react-cards` package and with the old `Card` before we're able to ship this. Hence why I'm not exporting the `Card` here yet and why this is under a `Card2` folder.

File: packages/react-cards/src/components/Card2/CardBase.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+export const CardBase = compose<'div', CardProps, CardProps, {}, {}>(
+  (props, ref, options) => {
+    const { state } = options;
+    const { slots, slotProps } = mergeProps<CardProps, CardProps, CardSlots, CardSlotProps>(state, options);
+
+    const { children } = state;
+
+    return (
+      <slots.root ref={useMergedRefs(ref)} {...slotProps.root}>
```
Comment (layershifter) [inline packages/react-cards/src/components/Card2/CardBase.tsx:15]:
```suggestion
      <slots.root ref={ref} {...slotProps.root}>
```

?

File: packages/react-cards/src/components/Card2/useCardBehavior.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { getCode, EnterKey, SpacebarKey } from '@fluentui/keyboard-key';
+import { CardProps, CardState } from './Card.types';
+
+export const useCardBehavior = (props: CardProps): Partial<CardState> => {
+  const { disabled, onClick: _onClick, onKeyDown: _onKeyDown } = props;
+
+  let onClick: ((ev: React.MouseEvent<HTMLDivElement>) => void) | undefined = undefined;
+  if (_onClick) {
```
Comment (layershifter) [inline packages/react-cards/src/components/Card2/useCardBehavior.ts:8]: Is it a premature optimization?...

File: packages/react-cards/src/components/Card2/useCardBehavior.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        const eventCode = getCode(ev);
+        if (eventCode === EnterKey || eventCode === SpacebarKey) {
+          onClick(ev as any);
+        }
+      }
+    }
+  };
+
+  return {
+    'aria-disabled': disabled,
```
Comment (layershifter) [inline packages/react-cards/src/components/Card2/useCardBehavior.ts:30]:
```suggestion
    'aria-disabled': disabled,
    role: 'group',
```

In Northstar there is `role` attr, see `cardBehavior`. Should we reuse it?

File: packages/react-cards/src/components/Card2/useCardBehavior.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    tabIndex: disabled ? undefined : 0,
+    ...props,
+    onClick,
+    onKeyDown,
```
Comment (layershifter) [inline packages/react-cards/src/components/Card2/useCardBehavior.ts:34]: `Card`s in Northstar are not focusable/actionable by default, only with `cardFocusableBehavior`. Is it intentional?

Comment (khmakoto) [reply packages/react-cards/src/components/Card2/useCardBehavior.ts:34]: We can make them focusable only when `onClick` has been provided. I'll make the change.

File: packages/react-cards/src/components/Card2/useCardBehavior.ts (hunk 3 above)
Comment (khmakoto) [reply packages/react-cards/src/components/Card2/useCardBehavior.ts]: Added

File: packages/react-cards/src/next/Card.types.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+export interface CardState extends CardProps {
+  cardRef: React.RefObject<HTMLDivElement>;
+}
+
+export interface CardSlots extends BaseSlots {}
+
+export type CardSlotProps = SlotProps<CardSlots, CardProps, React.HTMLAttributes<HTMLDivElement>>;
+
+export type CardTokens = ColorTokenSet & {};
```
Comment (dzearing) [inline packages/react-cards/src/next/Card.types.ts:55]: in my variants PR i am renaming `ColorTokenSet` to just `ColorTokens`. Just a heads up. If you merge first here, i'll get this updated.

**Review by dzearing — APPROVED** [review]

File: packages/react-cards/package.json [hunk 6]
```diff
@@ -48,10 +49,20 @@
     "react-test-renderer": "^16.3.0"
   },
   "dependencies": {
+    "@fluentui/keyboard-key": "^0.2.12",
+    "@fluentui/react-compose": "^0.19.3",
+    "@fluentui/react-theme-provider": "^0.13.3",
+    "@fluentui/theme": "^0.3.2",
     "@microsoft/load-themed-styles": "^1.10.26",
+    "@uifabric/azure-themes": "^7.5.13",
```
Comment (dzearing) [inline packages/react-cards/package.json:57]: wait, this should not be a dependency.

Comment (dzearing) [reply packages/react-cards/package.json:57]: check the others as well.

Comment (khmakoto) [reply packages/react-cards/package.json:57]: Spoke about this offline with David. A lot of this cleanup is already being done in #15132.

**Pull request merged**